pcap_linux.go

```diff
@@ -203,15 +221,15 @@ func (h *Handle) readPacketDataMmap() ([]captured, error) {
 			}
 			if sockOptVal == int(syscall.ENETDOWN) {
 				logger.Errorf("interface %s is down, marking as closed and returning", h.iface)
-				atomic.StoreUint32(&h.isOpen, 0)
+				atomic.StoreUint32(&h.state, closed)
```

Comment:
It seems a bit odd (but not necessarily causing bugs) to have both the reader here and the closer setting the state to closed. It would seem more logical to have the reader, when it sees errors like this, mark the state with some new error/failed/gone state, and then perhaps rename the cancelling to closing.

Thus the open -> cancelling/closing -> closed always comes from the Close() and the reader does open -> reading -> (open or gone) and the Close() will transition gone -> closed.
I don't know if the additional state makes it easier for others to understand.

Reply:
I can go either way. @milan-zededa thoughts?

Reply:
I have updated the patch and the state machine such that each state can be set only either by Writer (`OpenLive()` or `Close()`) or by Reader, not by both. I have introduced `gone` state but kept `canceling` for more clarity.